### PR TITLE
remove openai key assertion, enable alternate embedding models

### PR DESCRIPTION
**Pull request by rbren**

### CODE REVIEW

File: agenthub/langchains_agent/utils/memory.py [hunk 1]
```diff
@@ -1,18 +1,42 @@
...
+    embed_model = OllamaEmbedding(
+        model_name="llama2",
+        base_url=os.getenv("LLM_EMBEDDING_MODEL_BASE_URL", "http://localhost:8000"),
+        ollama_additional_kwargs={"mirostat": 0},
+    )
+elif embedding_strategy == "local":
+    from llama_index.embeddings.huggingface import HuggingFaceEmbedding
+    embed_model = HuggingFaceEmbedding(
+        model_name="BAAI/bge-small-en-v1.5"
+    )
```
Comment (RohitX0X) [inline agenthub/langchains_agent/utils/memory.py:26]: I have included another embedding model using azureopenai , raised PR for this branch, Let me know your review on it
thanks

Comment (rbren) [reply agenthub/langchains_agent/utils/memory.py:26]: Merged!

I haven't used custom embeddings much. Can you help me understand if we should have separate env vars for base_url, api_key, etc? Or should they be the same as for the core LLM?

Comment (rbren) [reply agenthub/langchains_agent/utils/memory.py:26]: Let's go with the smaller var for now--we can get more specific later

File: agenthub/langchains_agent/utils/memory.py [hunk 2]
```diff
@@ -1,18 +1,51 @@
...
+
+embedding_strategy = os.getenv("LLM_EMBEDDING_MODEL", "openai")
+
+# TODO: More embeddings: https://docs.llamaindex.ai/en/stable/examples/embeddings/OpenAI/
+# There's probably a more programmatic way to do this.
+if embedding_strategy == "llama2":
+    from llama_index.embeddings.ollama import OllamaEmbedding
+    embed_model = OllamaEmbedding(
+        model_name="llama2",
+        base_url=os.getenv("LLM_EMBEDDING_MODEL_BASE_URL", "http://localhost:8000"),
```
Comment (rbren) [inline agenthub/langchains_agent/utils/memory.py:19]:
```suggestion
        base_url=os.getenv("LLM_BASE_URL", "http://localhost:8000"),
```

File: agenthub/langchains_agent/utils/memory.py [hunk 3]
```diff
@@ -1,18 +1,51 @@
...
+        model="text-embedding-ada-002",
+        deployment_name=os.getenv("LLM_DEPLOYMENT_NAME"),
+        api_key=os.getenv("LLM_API_KEY"),
+        azure_endpoint=os.getenv("LLM_BASE_URL"),
+        api_version=os.getenv("LLM_API_VERSION"),                 
+    )
+else:
+    from llama_index.embeddings.openai import OpenAIEmbedding
+    embed_model = OpenAIEmbedding(
+        base_url=os.getenv("LLM_EMBEDDING_MODEL_BASE_URL"),
```
Comment (rbren) [inline agenthub/langchains_agent/utils/memory.py:39]:
```suggestion
        base_url=os.getenv("LLM_BASE_URL"),
```

File: requirements.txt [hunk 4]
```diff
@@ -16,3 +16,5 @@ langchain-community
 llama-index
 llama-index-vector-stores-chroma
 chromadb
+llama-index-embeddings-huggingface
+llama-index-embeddings-azure-openai
```
Comment (rbren) [inline requirements.txt:20]:
```suggestion
llama-index-embeddings-azure-openai
llama-index-embeddings-ollama
```

**Comment by rbren** [comment]
This is confirmed working by one of the users hitting the issue. Going to get this in as we've had several reports

**Pull request merged**